app/scripts/lib/accounts/BalancesController.ts

```diff
@@ -311,9 +349,15 @@ export class BalancesController extends BaseController<
       // Nothing to do here for EVM accounts
       return;
     }
+    const updateTimes = {
+      [BtcAccountType.P2wpkh]: BTC_AVG_BLOCK_TIME,
+      [SolAccountType.DataAccount]: SOLANA_AVG_BLOCK_TIME,
+    };
 
-    this.#tracker.track(account.id, BTC_AVG_BLOCK_TIME);
-    // NOTE: Unfortunately, we cannot update the balance right away here, because
+    const updateTime =
+      updateTimes[account.type as keyof typeof updateTimes] ||
+      SOLANA_AVG_BLOCK_TIME;
+    this.#tracker.track(account.id, updateTime); // NOTE: Unfortunately, we cannot update the balance right away here, because
```

Comment:
Good catch, but actually it should uses the same values, at least for Bitcoin. Since the block time is much higher, we use this "trick" to wait `blockTime / 2` to minimize the downtime.

So actually this was an error 😅

Could you re-use `#getBlockTimeFor` here too then? (that does not change anything for Solana block time anyway).